### PR TITLE
Fix bug 1215109. Added unchanged filter to project search.

### PR DESCRIPTION
**Pull request by jotes**
Hi @mathjazz @Osmose  could you review? Whole fix will need more tweaks in UI, i guess. However, i've added some tests for searching through entities. I hope i didn't break anything.

### CODE REVIEW

**Comment by Osmose** [comment]
This one is all @mathjazz :D

File: pontoon/base/templates/part_selector.html [hunk 1]
```diff
@@ -66,31 +68,16 @@
...
-          </li>
-        </ul>
+        {{ Search.options((
+          ('sources', 'Source strings', true),
+          ('translations', 'Translations', true),
+          ('comments', 'Comments', false),
+          ('keys', 'Keys', false),
+          ('casesensitive', 'Case sensitive', false),
+          ('unchanged', 'Unchanged', false),
+        ), request)}}
```
Comment (mathjazz) [inline pontoon/base/templates/part_selector.html]: Beautiful!

**Comment by mathjazz** [comment]
Nice work, @jotes! The code is simple and straightforward.

But I find the UI confusing. Previously available options in the list are joined with OR, whereas "Unchanged" is joined with AND. We should also add all other filters available in the sidebar search, which will make things even more confusing.

We've been talking about Search redesign for a while, so this might be the right time to re-think it.

@Osmose I wonder how far have you come with the entity list optimization (bug 1216546), because these two bugs are related. I agree with your proposal we should only have one search field. What if we remove the one in the resource menu completely, add "All resource" item to the resource menu and then only keep the search in the sidebar?

**Comment by Osmose** [comment]
> @Osmose I wonder how far have you come with the entity list optimization (bug 1216546), because these two bugs are related. I agree with your proposal we should only have one search field. What if we remove the one in the resource menu completely, add "All resource" item to the resource menu and then only keep the search in the sidebar?

I haven't really done anything touching search yet. My current focus was simply refactoring the view and improving the backend performance of get-entities in the average case.

I think I like your idea, overall.

**Comment by jotes** [comment]
Personally i thought about adding checkboxes on the right side on the input than below or simple label "filter by" above case-sensitive and unchanged.

**Pull request closed without merging**